dep-tree-da/README.md

```diff
@@ -0,0 +1,113 @@
+recruiting contact know what timeline works for you.
+
+## Requesting an Exemption
+
+The purpose of this assignment is to help us better understand your
+abilities as a data engineer. In addition, we may also use the work you
+produce as part of a pairing session during our hiring process.
+
+You may already have some publicly available work which would help us on both
+these fronts. If you do, let us know and we will take a look. We’re looking
+for something with the following characteristics:
+
+* This must be your project, meaning that you wrote most of the code and you
+  are the project's primary maintainer.
+* It should be something that is neither trivially small nor overwhelmingly
+  large. We’re looking for something that ranges from 500-2000(ish) lines of
+  code. Anything much smaller won’t have enough complexity for us to evaluate
+  your engineering skill. Anything much larger is just overwhelming for us to
+  evaluate and pair with you on.
+* There should be something in the project that you’d like to work on during a
+  pairing session. This could be fixing a bug, adding a small feature, doing a
+  refactoring, adding tests, etc. We don’t have to finish this work during the
+  session, but it should be something simple enough that we can really dig
+  into it in the space of 90 minutes.
+
+Please let your recruiting contact at ActiveState know if you have something
+you think is appropriate. We'll get back to you promptly to let you know if it
+is, in which case you can skip the homework assignment entirely.
+
+## Assignment
+
+We have provided some sample data in the form of the `META.json` files that
+accompany most Perl libraries uploaded to CPAN. See
+https://metacpan.org/pod/CPAN::Meta::Spec for documentation on this format.
+
+This data does not contain circular references and you do not need to handle
+those in your program.
+
+For this assignment, the only important key in this metadata is the
+**prereqs** key.
+
+The files are located under `data/` in a directory tree where each
+subdirectory corresponds to the name of a Perl distribution, like “DateTime”,
+"Eval-Closure", or "Module-Runtime".
+
+There are also some additional support files under `data/`. The first is a
+`data/module-distro-map.json` that maps Perl module names to Perl distribution
+names. Note that the `META.json` files belong to a distribution (distro), but
+the prereqs themselves always specify individual modules. A distro can contain
+one or more modules. This means you will need to translate from module names
+to distro names when you are constructing your dependency tree.
+
+All of the files you need are in the `data/` directory. You should not need to
+fetch additional data from CPAN or anywhere else on the Internet.
+
+The other is a `data/core-modules.json` file. This contains an array of
+modules which are shipped with the Perl core.
+
+Your assignment requires reading this metadata and producing fully resolved
+dependency trees.
```

Comment:
This might be at odds with the paragraph after the next one; perhaps merge them or remove this one?